core/src/com/unciv/logic/civilization/diplomacy/DiplomacyManager.kt

```diff
@@ -136,35 +136,28 @@ class DiplomacyManager() {
         if(civInfo.isPlayerCivilization())
             return otherCiv().getDiplomacyManager(civInfo).relationshipLevel()
 
-        if(civInfo.isCityState()){
-            if (influence<=-60) return RelationshipLevel.Unforgivable
-            if (influence<=-30) return RelationshipLevel.Enemy
+        if(civInfo.isCityState()) {
+            if(influence <= -60) return RelationshipLevel.Unforgivable
+            if(influence <= -30 || civInfo.isAtWarWith(otherCiv())) return RelationshipLevel.Enemy
 
-            if(civInfo.isAtWarWith(otherCiv()))
-                return RelationshipLevel.Enemy // See below, same with major civs
-
-            if(influence>=60) return RelationshipLevel.Ally
-            if(influence>=30) return RelationshipLevel.Friend
+            if(influence >= 60) return RelationshipLevel.Ally
+            if(influence >= 30) return RelationshipLevel.Friend
             return RelationshipLevel.Neutral
         }
 
         // not entirely sure what to do between AI civs, because they probably have different views of each other,
         // maybe we need to average their views of each other? That makes sense to me.
 
         val opinion = opinionOfOtherCiv()
-        if(opinion<=-80) return RelationshipLevel.Unforgivable
-        if(opinion<=-40) return RelationshipLevel.Enemy
-
-        // This is here because when you're at war you can either be enemy OR unforgivable,
-        // depending on the opinion
-        if(civInfo.isAtWarWith(otherCiv()))
-            return RelationshipLevel.Enemy
-
-        if(opinion<=-15) return RelationshipLevel.Competitor
-        if(opinion>=80) return RelationshipLevel.Ally
-        if(opinion>=40) return RelationshipLevel.Friend
-        if(opinion>=15) return RelationshipLevel.Favorable
-        return RelationshipLevel.Neutral
+        return when {
+            opinion >= 80  -> RelationshipLevel.Ally
+            opinion >= 40  -> RelationshipLevel.Friend
+            opinion >= 15  -> RelationshipLevel.Favorable
+            opinion <= -80 -> RelationshipLevel.Unforgivable
+            opinion <= -40 || civInfo.isAtWarWith(otherCiv()) -> RelationshipLevel.Enemy  /* During wartime, the estimation in which you are held may be enemy OR unforgivable */
```

Comment:
The negative checks need to be first, so that if you're at war but has good relations before you'll still be enemies

Reply:
Does that work?

Reply:
Yup!